bin/check-beanstalkd.rb

```diff
@@ -85,21 +85,27 @@ def acquire_beanstalkd_connection
   end
 
   def run
-    stats = acquire_beanstalkd_connection.tubes[config[:tube].to_s].stats
-    message 'All queues are healthy'
+    begin
+      stats = acquire_beanstalkd_connection.tubes[config[:tube].to_s].stats
+      message 'All queues are healthy'
 
-    warns, crits, msg = check_queues(stats)
-    msg.join("\n")
+      warns, crits, msg = check_queues(stats)
+      msg.join("\n")
 
-    if crits.size > 0 # rubocop:disable Style/ZeroLengthPredicate
-      message msg
-      critical
-    end
+      if crits.size > 0 # rubocop:disable Style/ZeroLengthPredicate
+        message msg
+        critical
+      end
 
-    if warns.size > 0 # rubocop:disable Style/ZeroLengthPredicate
-      message msg
-      warning
+      if warns.size > 0 # rubocop:disable Style/ZeroLengthPredicate
+        message msg
+        warning
+      end
+
+    rescue
```

Comment:
Ideally it is best to only rescue specific errors and not `StandardError` what was the actual error you got?

Reply:
this : 

```Check failed to run: Response failed with: NOT_FOUND, ["/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/beaneater-0.3.3/lib/beaneater/connection.rb:116:in `parse_response'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/beaneater-0.3.3/lib/beaneater/connection.rb:55:in `block in transmit'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/beaneater-0.3.3/lib/beaneater/connection.rb:49:in `synchronize'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/beaneater-0.3.3/lib/beaneater/connection.rb:49:in `transmit'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/beaneater-0.3.3/lib/beaneater/pool.rb:65:in `block (2 levels) in transmit_to_all'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/beaneater-0.3.3/lib/beaneater/pool.rb:141:in `safe_transmit'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/beaneater-0.3.3/lib/beaneater/pool.rb:65:in `block in transmit_to_all'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/beaneater-0.3.3/lib/beaneater/pool.rb:63:in `map'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/beaneater-0.3.3/lib/beaneater/pool.rb:63:in `transmit_to_all'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/beaneater-0.3.3/lib/beaneater/pool_command.rb:28:in `transmit_to_all'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/beaneater-0.3.3/lib/beaneater/tube/record.rb:96:in `stats'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-plugins-beanstalk-1.0.0/bin/check-beanstalkd.rb:88:in `run'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-plugin-1.4.2/lib/sensu-plugin/cli.rb:58:in `block in <class:CLI>'"]```

Reply:
I think a `rescue Beaneater::UnexpectedResponse` might do the trick: https://github.com/beanstalkd/beaneater/blob/c72df8540b3dd55dec084309d488efa508718f52/test/errors_test.rb#L7 if not we can try specifically resucing `Beaneater::NotFoundError`